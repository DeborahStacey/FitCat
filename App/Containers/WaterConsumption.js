import React from 'react'
import { AsyncStorage, Text, TextInput, ScrollView, View } from 'react-native'
import { Actions as NavigationActions } from 'react-native-router-flux'
import { default as StorageKeys } from '../Config/StorageKeys'
import RoundedButton from '../Components/RoundedButton'
import { default as WellCatManager } from '../Services/WellCatManager'
import I18n from 'react-native-i18n'

import styles from './Styles/ConsumptionStyles'

/*
Name
Brand OPTIONAL
Description OPTIONAL
*/

export default class WaterConsumption extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      formCupsText: ''
    }
  }

  submitWaterRecord = async () => {
    var inputWater = parseFloat(this.state.formCupsText)
    if (!inputWater > 0) {
      return
    }
    var currentPetId = await AsyncStorage.getItem(StorageKeys.CAT_ID)
    WellCatManager.submitWaterRecord(currentPetId, inputWater)
  }

  render () {
    return (
      <View style={styles.mainContainer}>
        <ScrollView style={styles.container}>
          <View style={styles.paddedContainer}>
            <Text style={styles.sectionText}>
              {I18n.t('cups_water_propt')}
            </Text>
            <TextInput
              style={{height: 40, borderColor: 'gray', borderWidth: 1, paddingLeft: 10, marginBottom: 15}}
              keyboardType={'numeric'}
              placeholder='Cups'
              onChangeText={(cups) => this.setState({
                formCupsText: cups
              })} />
            <RoundedButton onPress={() => this.submitWaterRecord()}>
              {I18n.t('submit_water_record')}
            </RoundedButton>

            <RoundedButton onPress={() => NavigationActions.viewWaterConsumption()}>
              {I18n.t('view_water_consumption')}
            </RoundedButton>
          </View>
        </ScrollView>
      </View>
    )
  }
}
