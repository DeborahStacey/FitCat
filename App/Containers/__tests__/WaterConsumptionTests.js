import React from 'react'
import WaterConsumption from '../WaterConsumption'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

it('renders the water consumption page correctly', () => {
  const tree = renderer.create(
    <WaterConsumption />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
