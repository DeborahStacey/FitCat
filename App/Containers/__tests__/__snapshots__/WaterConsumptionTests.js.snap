exports[`test renders the water consumption page correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F1F5FF",
      "flex": 1,
      "marginTop": 64,
    }
  }>
  <ScrollView
    style={
      Object {
        "flex": 1,
        "paddingTop": 10,
      }
    }>
    <View
      style={
        Object {
          "padding": 25,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#2C457F",
            "fontWeight": "bold",
            "marginVertical": 5,
            "textAlign": "center",
          }
        }>
        [missing \"en.cups_water_propt\" translation]
      </Text>
      <TextInput
        keyboardType="numeric"
        onChangeText={[Function]}
        placeholder="Cups"
        style={
          Object {
            "borderColor": "gray",
            "borderWidth": 1,
            "height": 40,
            "marginBottom": 15,
            "paddingLeft": 10,
          }
        } />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#466ECC",
            "borderRadius": 5,
            "height": 45,
            "justifyContent": "center",
            "marginHorizontal": 25,
            "marginVertical": 10,
            "opacity": 1,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginHorizontal": 5,
              "marginVertical": 10,
              "textAlign": "center",
            }
          }>
          [MISSING \"EN.SUBMIT_WATER_RECORD\" TRANSLATION]
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#466ECC",
            "borderRadius": 5,
            "height": 45,
            "justifyContent": "center",
            "marginHorizontal": 25,
            "marginVertical": 10,
            "opacity": 1,
          }
        }
        testID={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginHorizontal": 5,
              "marginVertical": 10,
              "textAlign": "center",
            }
          }>
          [MISSING \"EN.VIEW_WATER_CONSUMPTION\" TRANSLATION]
        </Text>
      </View>
    </View>
  </ScrollView>
</View>
`;
